### ISSUE TITLE
Improve navigational links

### ISSUE THREAD

**Issue by hseritt** [hseritt/repositext#2]
We need to improve on the navigational links. Currently we have:

Profile
Admin
About

I'm not sure we need an about link in navigational but it is only there for look and feel at the moment.

We probably ought to implement authentication so that "Profile" has some meaning. The "Admin" section would link to an admin console (maybe more for the Repositext app than the Django admin -- though it could link to the Django admin and we simply improve it).

**Comment by mridubhatnagar** [comment]
In my opinion navigational links on 

/docweb/ - Profile, Admin, Logout (Logout because this page would only be visible when user is logged in).

/user/home/ - Login, Logout. (Default page i.e home page can be the about page)

1. Profile and authentication can be separate. Otherwise, it would get confusing. In profile, we can display the user's profile only. In profile, we can display fields of extended user model.
2. Have a separate nav-link for logout.
3. Admin can take the user to the admin panel.
4. About can be on the home page. Inside a jumbotron, some content that tells about the project.

**Comment by hseritt** [comment]
Hi Mridu!

I've had a good look at your work. And thanks very much for the work you've
put into it. You seem to know what you're doing and are making a valuable
contribution.

But, I think we need to go into a different direction at least for now. I
realize my description in the issue was not as comprehensive as it should
have been. Apologies for that. You're on the right track for the login
piece but consider the following:

We want to keep the login piece inside of the docweb app. The docweb app is
the actual app that will handle *all* of the UI (at least for now -- we can
segment it later but for now we're using apps.docweb).

Bootstrap does look better but as far as css is concerned, we want to
continue to use the styles.css for now. We'll change that eventually. We
may use bootstrap or not.

We also want to keep the same header even on the login page.

Also, the functionality for logout should redirect the user to a login page
and not to a blank one.

We also do not want to set up registration at this point especially since
this is more of a deliverable or shrink-wrap type of application -- that
is, one that we can package and someone can download and install and use.
They can then run it on-premise. But, we also may want to consider a cloud
version and registration will be necessary, so I can see that you're
thinking ahead. I appreciate that.

Do let me know what you think and no one else is working on this piece (I
realize it's not an easy one). So, I'll be happy and very grateful to
consider any ideas you come up with.

I really appreciate your help and look forward to your re-work on this if
you want to continue on.

Thanks!
Harlin

**Comment by mridubhatnagar** [comment]
Hello, Thanks for having a look at the work. I did go through the updated requirements you mentioned here https://github.com/hseritt/repositext/issues/3

1. Yes, I can change the current code structure and push the changes from users app to docweb web. Wondering, if by doing so would the code remain modular and reusable?

2. Yes, as far as bootstrap is concerned and syncing the UI is concerned. We need to add it to style.css. Though for that you have separately opened an issue https://github.com/hseritt/repositext/issues/11. So, I think a PR for this issue would resolve the CSS thing and can together be worked on. I commented on the same here https://github.com/hseritt/repositext/pull/19.

3. Did you take a pull after the most recent changes? Because this behavior seems to be strange. I tested it out on my local. On doing logout the user gets redirected to /user/home page. Which is not blank. And from here he/she can again login. As login, signup are shown on top right.

4. No worries. I can scrape of SignUp form completely. But, then everytime how to we plan to create a new user who is allowed to do a login?

5. Can you share a rough drawing/wireframe of the login page you are imaging? That would help me in coming up with the exact login page. 

6. Also, will login page be our new home page? Because if you decide to keep the current /docweb/ page as home page. Then no point of login. As repository gets visible to everyone. I was considering to redirect the user to /docweb/ page only when he/she is logged in.

Once you share the following. I can re-work and push the changes again.

**Comment by hseritt** [comment]
Hi Mridu,

1. It will be as much as it needs to be for this project at the moment.
It's not the finished product by any stretch. We'll refactor out should we
need to.

2. This one has been taken care of. I actually need to close it. I'll do so
now.

3. It's possible I may have seen it wrong.

4. The idea for this is to be a distributed app. Whoever installs it will
be able to add users individually as his enterprise requires. This is the
nature of enterprise software. But, we will add a registration feature to
it later. We want to do this in steps. Also, it's my fault for not
explaining this as well as I should have initially.

5. Imagine a plain page with the current header. Then, imagine a login form
right in the middle of it. This page should be accessed at /docweb/login/.
On login, it should redirect to index.html or docweb/ and on logout, it
should redirect to /docweb/login/.

6. See #5.

Hope this helps but let me know if you have any questions on any of it.

Thank you for your help!

-Harlin

**Comment by mridubhatnagar** [comment]
Cool. I'll make the changes.

I didn't understand point 6. What needs to be done?

**Comment by hseritt** [comment]
For #6, login page will not be a "home" page. It only maps to
/docweb/login/.

Repository will only be visible to those who login -- eventually. Try not
to overthink it. Just keep within the parameters please.

On its own it will become more complex. But only complex as necessary.

Hope that helps.

-Harlin

**Comment by mridubhatnagar** [comment]
Yes currently it is like this only. Maps to /docweb/login/. Once table issue is fixed. Will push the changes.
